tools/cli/go-whisk/whisk/client.go

```diff
@@ -322,51 +322,69 @@ func (c *Client) Do(req *http.Request, v interface{}, ExitWithErrorOnTimeout boo
 
 func parseErrorResponse(resp *http.Response, data []byte, v interface{}) (*http.Response, error) {
     Debug(DbgInfo, "HTTP failure %d + body\n", resp.StatusCode)
+
     errorResponse := &ErrorResponse{Response: resp}
     err := json.Unmarshal(data, errorResponse)
 
-    // If the body matches the error response format, return an error containing the response error information.
-    // If an whisk.error() response was received parse the application error. (#5) Otherwise, the body contents are
-    // unknown (#6).
+    // Determine if an application error was received (#5)
     if err == nil {
-        if errorResponse.Code == 0 && len(errorResponse.ErrMsg) == 0 {
-            Debug(DbgInfo, "Error code, or error message is null\n")
-            return parseWhiskErrorResponse(resp, data, v)
+        if (errorResponse.Code == 0 || len(errorResponse.ErrMsg) == 0) && resp.StatusCode == 502 {
```

Comment:
is this code unnecessarily complex?
on any error, irrespective of whether "code" exists, the response is 502, or the error response is json, should the handling just emit the response as is and not try to overly interpret it?

that is, if result is not 200, try to parse a json, if success, emit the response as json object, else it's a string so emit as string.